lms/models/public_id.py

```diff
@@ -0,0 +1,52 @@
+from base64 import urlsafe_b64encode
+from dataclasses import dataclass
+from uuid import uuid4
+
+from lms.models.region import Region
+
+
+@dataclass
+class PublicId:
+    """
+    Get a globally unique value with prefixes like 'us.' or 'ca.'.
+
+    This is useful if you only have the id, but don't know which region you
+    should be looking for it in and is the only id suitable for sharing
+    outside a single region LMS context.
+    """
```

Comment:
Splitting this out into it's own object separate from the mixin seems a bit much, but allows us to reason about public ids outside of an SQLAlchemy context.

For example, if you wanted to accept a public id string, and _then_ determine which model to get, you could use this class.

It also allows for some splitting up of the tests. Unfortunately the way these things compose together means we don't get an overall reduction in testing, as we tend to be forced by practicality to test the composed objects in a slightly black box way rather than our usual mocking.